exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -364,4 +401,35 @@ private static SchemaPlus rootSchema(SchemaPlus schema) {
     }
   }
 
+  private static class DrillRexBuilder extends RexBuilder {
+    private DrillRexBuilder(RelDataTypeFactory typeFactory) {
+      super(typeFactory);
+    }
+
+    @Override
+    public RexNode ensureType(
+        RelDataType type,
+        RexNode node,
+        boolean matchNullability) {
+      RelDataType targetType = type;
+      if (matchNullability) {
+        targetType = matchNullability(type, node);
+      }
+      if (targetType.getSqlTypeName() == SqlTypeName.ANY) {
+        return node;
+      }
+      if (!node.getType().equals(targetType)) {
+        if(!targetType.isStruct()) {
+          final RelDataType anyType = TypeInferenceUtils.createCalciteTypeWithNullability(
```

Comment:
Cast an expression to "any" type? why do we have to add this? On line 418, in Calcite, the code's intention is to skip CAST to "any", since it does not make sense to cast to "any". Why do you want to explicitly add this here?